tests/unit_tests/tools/test_flags.py

```
@@ -68,6 +69,129 @@ def test_remove_flags():
+    with pytest.raises(KeyError) as err:
+        _ = pf["does_not_exist"]
+    assert "Profile 'does_not_exist' is not defined" in str(err.value)
+
+    # Check that we get an exception if we try to inherit from a profile
+    # that does not exist
+    with pytest.raises(KeyError) as err:
+        _ = pf.define_profile("new_profile", "does_not_exist")
+    assert ("Inherited profile 'does_not_exist' is not defined."
+            in str(err.value))
+
+    # Test that inheriting from the default profile "" works
+    pf.define_profile("from_default", "")
+    assert pf._inherit_from["from_default"] == ""
+
+
+def test_profile_flags_inheriting():
+    '''Tests adding flags.'''
+    pf = ProfileFlags()
+    pf.define_profile("base")
+    assert pf["base"] == []
+    # And there should not be any inherited profile defined:
+    assert "base" not in pf._inherit_from
+
+    pf.add_flags("-base", "base")
+    assert pf["base"] == ["-base"]
+
+    pf.define_profile("derived", "base")
+    assert pf["derived"] == ["-base"]
+    assert pf._inherit_from["derived"] == "base"
+    pf.add_flags("-derived", "derived")
+    assert pf["derived"] == ["-base", "-derived"]
+
+    pf.define_profile("derived2", "derived")
+    assert pf["derived2"] == ["-base", "-derived"]
+    pf.add_flags("-derived2", "derived2")
+    assert pf["derived2"] == ["-base", "-derived", "-derived2"]
+
+
+def test_profile_flags_removing():
+    '''Tests adding flags.'''
+    pf = ProfileFlags()
+    pf.define_profile("base")
+    assert pf["base"] == []
+    pf.add_flags(["-base1", "-base2"], "base")
+    warn_message = "Removing managed flag '-base1'."
+    with pytest.warns(UserWarning, match=warn_message):
+        pf.remove_flag("-base1", "base")
+    assert pf["base"] == ["-base2"]
+
+    pf.add_flags(["-base1", "-base2"])
+    warn_message = "Removing managed flag '-base1'."
+    with pytest.warns(UserWarning, match=warn_message):
+        pf.remove_flag("-base1")
+    assert pf[""] == ["-base2"]
+
```

Would it be good to add a test here of removing flags that do not exist from a profile?

---

Yes. I am actually not sure if this is the 'right' behaviour (maybe it should raise an exception??) - leave it for now till we get more experience.